Kuadrant/authorino-operator
# Adds status conditions to the Authorino API definition

This PR adds [conditions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties) to the status block of the authorino API definition, this will provide a higher-level status reporting  for the authorino instances managed by the operator.

The PR also contains some code refactoring to the controller.

Closes https://github.com/Kuadrant/authorino-operator/issues/12

### Verification steps
1) Install and deploy the operator from this branch 
2) Create a simple instance of authorino
```bash
kubectl -n myapp apply -f -<<EOF
apiVersion: operator.authorino.kuadrant.io/v1beta1
kind: Authorino
metadata:
  name: authorino
spec:
  replicas: 1
  clusterWide: false
  listener:
    tls:
      enabled: false
  oidcServer:
    tls:
      enabled: false
EOF
```
3) Verify conditions were set to the status definition of the API
```yaml
status:
  conditions:
  - lastTransitionTime: "2021-12-06T14:19:12Z"
    reason: Provisioned
    status: "True"
    type: Ready
```
